src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -115,7 +115,24 @@ public string HttpVersion
         public IHeaderDictionary RequestHeaders { get; set; }
         public Stream RequestBody { get; set; }
 
-        public int StatusCode { get; set; }
+        private int _statusCode;
+        public int StatusCode
```

Comment:
I would probably just change that to something like:


    if (HasResponseStarted)
    {
        // We can no longer respond with a 500, so we simply close the connection.
        _requestProcessingStopping = true;
        return TaskUtilities.CompletedTask;
    }
    else
    {
        if (_requestRejected)
        {
            // 400 Bad Request
            StatusCode = 400;
            _keepAlive = false;
        }
        else
        {
            // 500 Internal Server Error
            StatusCode = 500;
        }

        // ...